modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorsBuilder.es.js

```diff
@@ -0,0 +1,286 @@
+											initialQuery={criteria.query}
+											inputId={criteria.inputId}
+											modelLabel={criteria.modelLabel}
+											onChange={this.onCriteriaChange}
+											onEditionToggle={this.onCriteriaEdit}
+											propertyKey={criteria.propertyKey}
+											supportedConjunctions={supportedConjunctions}
+											supportedOperators={supportedOperators}
+											supportedProperties={criteria.properties}
+											supportedPropertyGroups={this.props.propertyGroups.map(
+												p => ({
+													label: p.name,
+													value: p.propertyKey
+												})
+											)}
+											supportedPropertyTypes={supportedPropertyTypes}
+										/>
+										<div className="form-group">
+											<input
+												className="field form-control"
+												data-testid={criteria.inputId}
+												id={criteria.inputId}
+												name={criteria.inputId}
+												readOnly
+												type="hidden"
+												value={criteria.query}
+											/>
+										</div>
+									</React.Fragment>
+								);
+							}
+						)
+					}
+					<div className={'sheet-lg'}>
+						{
+							this.state.contributors &&
+							this.state.contributors.map(
+								(c, i) => {
+									return (i !== 0 && c.query !== '') ? ` ${this.state.conjunctionName} ${c.query}` : c.query;
+								}
+							)
+						}
+					</div>
+				</div>
+				<div className="criteria-builder-section-sidebar">
+					{
+						<CriteriaSidebar
+							propertyKey={selectedProperty && selectedProperty.propertyKey}
+							supportedProperties={selectedProperty && selectedProperty.properties}
+							title={
+								sub(
+									Liferay.Language.get('x-properties'),
+									[selectedProperty && selectedProperty.name]
+								)
+							}
+						/>
+					}
+				</div>
+			</div>
+		);
+	}
+}
+
+const conjuctions = PropTypes.shape(
+	{
+		label: PropTypes.string.isRequired,
+		name: PropTypes.string.isRequired
+	}
+);
+const initialContributor = PropTypes.shape(
+	{
+		conjunctionId: PropTypes.string.isRequired,
+		conjunctionInputId: PropTypes.string.isRequired,
+		initialQuery: PropTypes.string.isRequired,
+		inputId: PropTypes.string.isRequired,
+		propertyKey: PropTypes.string.isRequired
+	}
+);
+const operators = PropTypes.shape(
+	{
+		label: PropTypes.string.isRequired,
+		name: PropTypes.string.isRequired
+	}
+);
+const property = PropTypes.shape(
+	{
+		label: PropTypes.string.isRequired,
+		name: PropTypes.string.isRequired,
+		type: PropTypes.string.isRequired
+	}
+);
+const propertyGroup = PropTypes.shape(
+	{
+		name: PropTypes.string.isRequired,
+		properties: PropTypes.arrayOf(property),
+		propertyKey: PropTypes.string.isRequired
+	}
+);
+const propertyTypes = PropTypes.shape(
+	{
+		boolean: PropTypes.arrayOf(PropTypes.string).isRequired,
+		date: PropTypes.arrayOf(PropTypes.string).isRequired,
+		number: PropTypes.arrayOf(PropTypes.string).isRequired,
+		string: PropTypes.arrayOf(PropTypes.string).isRequired
+	}
+);
```

Comment:
Let's add new lines here. Constants usually get new lines in between them too, unless the constants are related like

```javascript
// no new line
const labelOne = 'one';
const labelTwo = 'two';
```